includes/checkout.php

```diff
@@ -43,284 +46,51 @@ function fastwc_get_cart_data() {
- *
- * @return array
- */
-function fastwc_get_products_to_hide_buttons() {
-	$fastwc_hidden_products = get_option( FASTWC_SETTING_HIDE_BUTTON_PRODUCTS );
-
-	if ( ! empty( $fastwc_hidden_products ) ) {
-		$fastwc_count_products = count( $fastwc_hidden_products );
-
-		for ( $i = 0; $i < $fastwc_count_products; $i++ ) {
-			$fastwc_hidden_products[ $i ] = (int) $fastwc_hidden_products[ $i ];
-		}
-	}
-
-	return $fastwc_hidden_products;
-}
-
-/**
- * Determine if the Fast PDP button should be hidden for a specific product.
- *
- * @return bool
- */
-function fastwc_should_hide_pdp_button_for_product() {
-	$fastwc_hidden_products = fastwc_get_products_to_hide_buttons();
-
-	if ( ! empty( $fastwc_hidden_products ) && is_product() ) {
-		// Check current product ID.
-		global $product;
-
-		$product_id = ! empty( $product ) ? $product->get_id() : 0;
-
-		if ( ! empty( $product_id ) && in_array( $product_id, $fastwc_hidden_products, true ) ) {
-			return true;
-		}
-	}
-
-	return false;
-}
-
-/**
- * Determine if the Fast cart button should be hidden for a specific product.
- *
- * @return bool
- */
-function fastwc_should_hide_cart_button_for_product() {
-	$fastwc_hidden_products = fastwc_get_products_to_hide_buttons();
-
-	if ( ! empty( WC()->cart ) ) {
-		$cart = WC()->cart->get_cart();
-
-		foreach ( $cart as $cart_item ) {
-			$product_id = ! empty( $cart_item['product_id'] ) ? $cart_item['product_id'] : 0;
-
-			if ( ! empty( $product_id ) && in_array( $product_id, $fastwc_hidden_products, true ) ) {
-				return true;
-			}
-		}
+function fastwc_maybe_render_cart_button( $template ) {
+	if ( fastwc_should_hide_cart_checkout_button() ) {
+		return;
 	}
 
-	return false;
+	fastwc_load_template( $template );
 }
 
-/**
- * Product detail page
- */
-
-/**
- * Detect if the product has any addons (Fast Checkout does not yet support these products).
- */
-function fastwc_wc_product_addon_start() {
-	// If the store has the addons plugin installed, then this hook will run on any PDP pages for products with addons.
-	// In this situation, we want to note this so that our hook that displays the button can instead hide the button.
-	global $fastwc_product_has_addons;
-	$fastwc_product_has_addons = true;
-}
-add_action( 'wc_product_addon_start', 'fastwc_wc_product_addon_start' );
-
-/**
- * Detect if the product is a grouped product (Fast Checkout does not yet support these products).
- */
-function fastwc_woocommerce_grouped_product_list_before() {
-	global $fastwc_product_is_grouped;
-	$fastwc_product_is_grouped = true;
-}
-add_action( 'woocommerce_grouped_product_list_before', 'fastwc_woocommerce_grouped_product_list_before' );
-
 /**
  * Inject Fast Checkout button after Add to Cart button.
  */
 function fastwc_woocommerce_after_add_to_cart_quantity() {
-	if ( fastwc_should_hide_checkout_button() ) {
+	if ( fastwc_should_hide_pdp_checkout_button() ) {
 		return;
 	}
 
 	fastwc_load_template( 'fast-pdp' );
```

Comment:
should we have a `maybe_render_pdp_button` function that we use here that is similar to the one you wrote for cart button?

Reply:
I added the `maybe_render_cart_button` because the same logic is used in multiple places, but for the sake of consistency I think it's a good idea to add the `maybe render_pdp_button` function as well. I'll get that added.

Reply:
I updated the function to be a more general `fastwc_maybe_render_checkout_button` and added a `$button_type` parameter.